### ISSUE TITLE
cuda hooks segfaulting

### ISSUE THREAD

**Issue by soumith** [pytorch/pytorch#531]
```
import torch
import torch.cuda
import torch.nn as nn
from torch.autograd import Variable

x = torch.FloatTensor(16, 30)
linear = nn.Linear(30, 20)

y = linear(Variable(x))
y_target = torch.FloatTensor(16, 20)

loss = nn.MSELoss()
def hook(layer, grad_input, grad_output):
    return grad_input * 2
loss.register_backward_hook(hook)

err = loss(y, Variable(y_target))
err.backward()
```
removing `loss.register_backward_hook(hook)` resolves it

**Comment by soumith** [comment]
reported by @yuandong-tian
